### PR TITLE
Make TsFileSequenceReader concurrent safety.

### PR DESCRIPTION
**Pull request by Beyyes**
Make TsFileSequenceReader concurrent safety and fix corresponding tests.

### CODE REVIEW

**Comment by MyXOF** [comment]
our jenkins pipelines cannot be triggered from a forked repo

File: iotdb/src/main/java/cn/edu/tsinghua/iotdb/query/timegenerator/EngineTimeGenerator.java [hunk 1]
```diff
@@ -43,4 +45,12 @@ public long next() throws IOException {
     @Override public Object getValue(Path path, long time) {
         return null;
     }
+
+    /**
+     * Close and remove all opened file streams by EngineTimeGenerator.
+     * This method is necessary in unit test which only created EngineTimeGenerator but not
```
Comment (jixuan1989) [inline iotdb/src/main/java/cn/edu/tsinghua/iotdb/query/timegenerator/EngineTimeGenerator.java]: ?

File: tsfile/src/main/java/cn/edu/tsinghua/tsfile/read/controller/MetadataQuerierByFileImpl.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public void loadChunkMetaDatas(List<Path> paths) throws IOException {
             TsDeviceMetadata tsDeviceMetadata = tsFileReader.readTsDeviceMetaData(index);
 
             // d1
-            for (ChunkGroupMetaData chunkGroupMetaData : tsDeviceMetadata.getChunkGroups()) {
+            for (ChunkGroupMetaData chunkGroupMetaData : tsDeviceMetadata.getChunkGroups()) {//TODO make this function better
```
Comment (Beyyes) [inline tsfile/src/main/java/cn/edu/tsinghua/tsfile/read/controller/MetadataQuerierByFileImpl.java]: TODO what?

File: tsfile/src/main/java/cn/edu/tsinghua/tsfile/file/header/PageHeader.java [hunk 3]
```diff
@@ -26,10 +27,10 @@ public static int calculatePageHeaderSize(TSDataType type) {
     }
 
     public int calculatePageHeaderSize() {
-        if(statistics!=null)
+        //if(statistics!=null)
```
Comment (Beyyes) [inline tsfile/src/main/java/cn/edu/tsinghua/tsfile/file/header/PageHeader.java]: remove these comments?

File: tsfile/src/main/java/cn/edu/tsinghua/tsfile/file/header/PageHeader.java [hunk 4]
```diff
@@ -76,11 +78,11 @@ public void setNumOfValues(int numOfValues) {
         return statistics;
     }
 
-    public void setStatistics(Statistics<?> statistics) {
+    /*public void setStatistics(Statistics<?> statistics) {
```
Comment (Beyyes) [inline tsfile/src/main/java/cn/edu/tsinghua/tsfile/file/header/PageHeader.java]: remove these comments?

**Pull request merged**